### PR TITLE
Added Job to Trigger a Release of the Docker Desktop Extension

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR adds a job to the production workflow that deploys MindsDB artifacts to trigger a new release of the Docker Desktop extension. This is done by triggering [this workflow](https://github.com/mindsdb/mindsdb-docker-extension/blob/main/.github/workflows/bump-mindsdb-version.yml) setup in the repository for the extension.

Fixes https://github.com/mindsdb/mindsdb-docker-extension/issues/6
Fixes https://github.com/mindsdb/mindsdb_private/issues/575

## Type of change

- [X] ⚡ New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB - N/A
- [X] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added - N/A

### CODE REVIEW

File: .github/workflows/build_deploy_prod.yml [hunk 1]
```diff
@@ -114,9 +114,31 @@ jobs:
...
+    concurrency:
+      group: deploy-prod
+      cancel-in-progress: false
+    steps:
+      - uses: FranzDiebold/github-env-vars-action@v2
+      - uses: convictional/trigger-workflow-and-wait@v1.6.5
+        with:
+          owner: mindsdb
+          repo: mindsdb-docker-extension
+          github_token: ${{ secrets.DD_EXT_REPO_DISPATCH_PAT_TOKEN }}
```
Comment (hamishfagg) [inline .github/workflows/build_deploy_prod.yml:135]: Can we use `REPO_DISPATCH_PAT_TOKEN` here? I have updated `REPO_DISPATCH_PAT_TOKEN` to allow access to the docker extension repo

**Review by hamishfagg — APPROVED** [review]

**Comment by hamishfagg** [comment]
Approved with above changes

File: .github/workflows/build_deploy_prod.yml [hunk 2]
```diff
@@ -114,9 +114,31 @@ jobs:
...
+  trigger_dd_extension_release:
+    # Trigger private repo to deploy to prod env
+    runs-on: mdb-dev
+    needs: docker_build
+    if: github.actor != 'mindsdbadmin'
+    environment:
+      name: prod
+    # We only want to run one deploy job for an env at a time
+    # Don't cancel in progress jobs because it may be for a different PR
+    concurrency:
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply .github/workflows/build_deploy_prod.yml]: Done.

File: .github/workflows/build_deploy_prod.yml (hunk 1 above)
Comment (MinuraPunchihewa) [reply .github/workflows/build_deploy_prod.yml]: Done.

**Pull request merged**